tests/unit/core/reducers/test_addons.js

```diff
@@ -32,42 +39,65 @@ describe(__filename, () => {
     const firstState = addons(undefined,
       loadAddons(createFetchAddonResult(fakeAddon).entities));
 
-    const anotherFakeAddon = { ...fakeAddon, slug: 'testing1234', id: 6401 };
+    const anotherFakeAddon = {
+      ...fakeAddon,
+      slug: 'testing1234',
+      id: 6401,
+    };
     const newState = addons(firstState,
       loadAddons(createFetchAddonResult(anotherFakeAddon).entities));
 
     const internalAddon = createInternalAddon(anotherFakeAddon);
-    expect(newState).toEqual({
-      ...firstState,
-      [anotherFakeAddon.slug]: internalAddon,
+    expect(newState.addons).toEqual({
+      ...firstState.addons,
       [anotherFakeAddon.id]: internalAddon,
     });
+    expect(newState.bySlug).toEqual({
+      ...firstState.bySlug,
+      [anotherFakeAddon.slug]: anotherFakeAddon.id,
+    });
+    expect(newState.byGUID).toEqual({
+      ...firstState.byGUID,
+      [anotherFakeAddon.guid]: anotherFakeAddon.id,
+    });
   });
 
-  it('stores all add-ons, indexed by id and slug', () => {
+  it('stores all add-ons, indexed by id', () => {
     const addonResults = [
       { ...fakeAddon, slug: 'first-slug', id: 123 },
       { ...fakeAddon, slug: 'second-slug', id: 456 },
     ];
     const state = addons(undefined,
       loadAddons(createFetchAllAddonsResult(addonResults).entities));
-    expect(Object.keys(state).sort())
-      .toEqual(['123', '456', 'first-slug', 'second-slug']);
+    expect(Object.keys(state.addons).sort()).toEqual(['123', '456']);
+  });
+
+  it('store all add-on slugs with their IDs', () => {
+    const addonResults = [
+      { ...fakeAddon, slug: 'first-slug', id: 123 },
+      { ...fakeAddon, slug: 'second-slug', id: 456 },
+    ];
+    const state = addons(undefined,
+      loadAddons(createFetchAllAddonsResult(addonResults).entities));
+    expect(state.bySlug).toEqual({
+      'first-slug': 123,
+      'second-slug': 456,
+    });
   });
 
   it('ignores empty results', () => {
     const addonResults = [];
     const state = addons(undefined,
       loadAddons(createFetchAllAddonsResult(addonResults).entities));
-    expect(Object.keys(state)).toEqual([]);
+    expect(Object.keys(state.addons)).toEqual([]);
```

Comment:
How is this equal to an array and not an [empty object](https://github.com/mozilla/addons-frontend/pull/3927/files#diff-21aa4f8ea4fba6406233649d0ce65beaR298)?

Reply:
because it is using `Object.keys()`?